### PR TITLE
lib: sms: fix assertion fail in at_cmd

### PR DESCRIPTION
**Pull request by arvinf**
When an SMS message is received calling `at_cmd_write` to acknowledge the PDU
from within the notification callback causes an `ASSERTION FAIL` in `at_cmd`
because it is being called from the socket thread.

e.g.
```
ASSERTION FAIL [k_current_get() != socket_tid] @ WEST_TOPDIR/nrf/lib/at_cmd/at_cmd.c:361
at_cmd deadlock: socket thread blocking self
```

The fix submits a work to the system work queue to acknowledge the PDU.

### CODE REVIEW

File: lib/sms/sms.c [hunk 1]
```diff
@@ -135,8 +136,18 @@ static int sms_cmt_notif_save(void)
 	return 0;
 }
 
+static void sms_ack_work(struct k_work *work)
+{
+	/* Acknowledge the SMS PDU in any case. */
```
Comment (bjda) [inline lib/sms/sms.c:141]: This comment can now be removed IMO, given the name and short nature of the function

**Review by bjda — COMMENTED** [review]
Makes sense to me. This is the kind of thing that assert was meant to catch.

One nitpick: the work item name `sms_work` is a bit too general.
I suggest renaming the handler function to `sms_ack` (alternatively `sms_ack_work_handler`), and rename the work item to `sms_ack_work`.

**Review by metc — APPROVED** [review]
LGTM. We can only receive one SMS at the time. The SMS has to be acknowledged to receive a new SMS, and this should be done quite "fast" or we will receive the same SMS again.

With the latest `at_cmd` driver, now all users of this module have to switch context to send new AT commands. Using a work queue and the system workqueue for this purpose works with me. I do not know if there is a more lightweight primitive for this though.

**Review by bjda — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
The CI failure is unrelated, merging.

**Pull request merged**